### ISSUE TITLE
Discovery+ no longer filtering

### ISSUE THREAD

**Issue by Jenron3000** [FrostCo/AdvancedProfanityFilter#550]
#### :bug: Discovery+ no longer muting audio

I have tried advanced and deep filtering options as well as changing from browser mute, to video mute, ect. 
Something changed because it has been working wonderfully. 
The captions are being filter but the audio doesn't mute
I've tried multiple browsers to no avail.
Something changed because it has been working wonderfully. 
Thank you so much for your work on this filter. 

#### :computer: Details

- **Device:**  MacBook Pro
- **OS:**  MacOS 12.7.1
- **Browser:**   Brave, Chrome, Firefox.
- **Browser Version:** Version 1.64.122 Chromium: 123.0.6312.122 (Official Build) (x86_64)]
- **Affected site(s):** Discoveryplus.com

**Comment by richardfrost** [comment]
Hi @Jenron3000, thanks for reporting the issue. I'm glad that you are enjoying the filter when it is working for you though! Hopefully we can get it fixed up again.

It looks like Discovery+ has changed quite a bit about their website, but I do have a potential update that will hopefully fix it. Could you give it a try and let me know how it works for you?

```json
{
  "www.discoveryplus.com": [
    {
      "className": "sc-iMWBiJ",
      "mode": "element",
      "subtitleSelector": "div.sc-ihgnxF",
      "tagName": "div"
    }
  ]
}
```

To use this config, paste it into the Custom Sites area (Options page > Audio tab) and click the blue "SAVE" button underneath it. Then, make sure to refresh the tab for the video site (if you have it open still) and it will load this new configuration.

**Comment by richardfrost** [comment]
Sorry @Jenron3000, I did some more testing with that config I shared earlier and it definitely has some issues. This one should work much better:

```json
{
  "www.discoveryplus.com": [
    {
      "apfCaptions": true,
      "apfCaptionsSelector": "div#overlay-root",
      "displaySelector": "div[data-testid=caption_renderer_overlay]",
      "mode": "watcher",
      "subtitleSelector": "div[data-testid=caption_renderer_overlay] div.sc-iMWBiJ div.sc-ihgnxF",
      "videoSelector": "div[data-testid=caption_renderer_overlay] video"
    }
  ]
}
```

**Comment by Jenron3000** [comment]
I tried dropping that code in and i still only blocks the caption and doesn't mute the audio. Should it change the code in the "supported sites config" list? after i save the code it doesn't change it.

**Comment by richardfrost** [comment]
Thanks for getting back to me @Jenron3000. Good question, but it won't change the "support sites config" in the list, those are just the built-in configs and are read-only.

I'm sorry, I just noticed both a typo and an error in the 2nd config I shared with you. Could you please try this one?

```json
{
  "www.discoveryplus.com": [
    {
      "apfCaptions": true,
      "apfCaptionsSelector": "div#overlay-root",
      "displaySelector": "div[data-testid=caption_renderer_overlay]",
      "mode": "watcher",
      "subtitleSelector": "div[data-testid=caption_renderer_overlay] div.sc-iMWBiJ div.sc-ihgnxF",
      "videoSelector": "div[data-testid=video-container] video"
    }
  ]
}
```

Also, please make sure the site is set to normal (not Advanced or Deep).

If this one still doesn't work, can  you share an example video so I can test it? Thanks!

**Comment by richardfrost** [comment]
@Jenron3000 When you get a chance let me know if this latest version works for you. Sorry again for the errors in the previous configs.

**Comment by Jenron3000** [comment]
Yes it’s working excellently now with discovery plus! Thank you! On May 1, 2024, at 11:51 PM, Richard Frost ***@***.***> wrote:﻿
@Jenron3000 When you get a chance let me know if this latest version works for you. Sorry again for the errors in the previous configs.

**Comment by richardfrost** [comment]
Thanks for letting me know @Jenron3000.

This has been included in version [3.5.0](https://github.com/FrostCo/AdvancedProfanityFilter/releases/tag/3.5.0) of the extension. For best results, make sure to remove any custom config used for testing like this once it has been included in the official update so you can automatically get future updates.
